### PR TITLE
Clean up Version tests

### LINKED ISSUE #6286: Cleanup System.Runtime.Tests
This is a meta-issue to track progress on cleaning up System.Runtime.Tests. I dumped a massive PR (#5490) which I'm splitting up into smaller ones
The following test files should be cleaned up, to add additional coverage, and follow test guidelines better:

- [x] String (#6296)
- [x] StringBuilder (#6295)
- [ ] Number primitives: byte, int, long etc.
- [ ] Floating primitives: float, double
- [x] Decimal (#6297)
- [x] Array (#6293)
- [ ] Array Segment
- [ ] Buffer
- [ ] DateTime
- [ ] DateTimeOffset
- [ ] TimeSpan
- [ ] Char
- [ ] DefaultValue and EditorBrowsable
- [ ] System.IO exceptions
- [x] Activator (#6294)
- [ ] Attribute
- [ ] System exceptions
- [ ] GC
- [x] Version (#6298)
- [x] Guid (#6299)
- [ ] Nullable
- [ ] Lazy
- [x] Pointers - IntPtr and UIntPtr (#6300)

After this:
- Add missing namespaces
- Rename tests as per #4535

### PR DESCRIPTION
**Pull request by hughbe**

### CODE REVIEW

File: src/System.Runtime/tests/System/Version.cs [hunk 1]
```diff
@@ -3,211 +3,263 @@
...
     {
-        VerifyConstructor(0, 0, 0, 0);
-        VerifyConstructor(2, 3, 4, 7);
-        VerifyConstructor(2, 3, 4, 65535);
-        VerifyConstructor(2, 3, 4, 65536);
-        VerifyConstructor(2, 3, 4, 32767);
-        VerifyConstructor(2, 3, 4, 32768);
-        VerifyConstructor(2, 3, 4, 2147483647);
-        VerifyConstructor(2, 3, 4, 2147450879);
-        VerifyConstructor(2, 3, 4, 2147418112);
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/Version.cs]: Why did you remove these last six cases?

File: src/System.Runtime/tests/System/Version.cs [hunk 2]
```diff
@@ -3,211 +3,263 @@
...
     }
 
     [Fact]
-    public void CompareToObject()
+    public static void TestCompareTo_Invalid()
     {
-        var x = (IComparable)new Version(1, 2);
-        Assert.Equal(1, x.CompareTo(null));
-        Assert.Equal(0, x.CompareTo(new Version(1, 2)));
-        Assert.True(x.CompareTo(new Version(1, 3)) < 0);
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/Version.cs]: Why did you remove this case?

File: src/System.Runtime/tests/System/Version.cs (hunk 1 above)
Comment (hughbe) [inline src/System.Runtime/tests/System/Version.cs]: I didn't. They have been subsumed into TestCtor_Int_Int....

File: src/System.Runtime/tests/System/Version.cs (hunk 2 above)
Comment (hughbe) [inline src/System.Runtime/tests/System/Version.cs]: I didn't: https://github.com/hughbe/corefx/commit/7c96d06f7f55f98a78f5e76e042b99c3153a0f94#diff-e920d148570a23dc60abba6cf8bbeddcR78

File: src/System.Runtime/tests/System/Version.cs [hunk 3]
```diff
@@ -3,211 +3,263 @@
...
+        Assert.Throws<ArgumentNullException>("v1", () => nullVersion <= testVersion); // V1 is null
     }
 
-    [Fact]
-    public void TestCompareTo()
+    private static IEnumerable<object[]> EqualsTestData()
     {
-        var x = new Version(1, 2);
-        Assert.Equal(0, x.CompareTo(new Version(1, 2)));
-        Assert.True(x.CompareTo(new Version(1, 3)) < 0);
```
Comment (stephentoub) [inline src/System.Runtime/tests/System/Version.cs]: Same question

File: src/System.Runtime/tests/System/Version.cs (hunk 2 above)
Comment (stephentoub) [inline src/System.Runtime/tests/System/Version.cs]: I'm searching for "1, 3", and I don't see it anywhere in your new version.  Where is it?  The one you linked to is ```yield return new object[] { new Version(1, 2), null, 1 };```

File: src/System.Runtime/tests/System/Version.cs (hunk 1 above)
Comment (hughbe) [inline src/System.Runtime/tests/System/Version.cs]: Actually I lied. I did remove them in part. Would you like me to add them back

Comment (stephentoub) [inline src/System.Runtime/tests/System/Version.cs]: I'm not following.  All I see those testing is int.MaxValue, not these other values (like short.MaxValue, ushort.MaxValue, etc.)... where are those?

File: src/System.Runtime/tests/System/Version.cs (hunk 2 above)
Comment (hughbe) [inline src/System.Runtime/tests/System/Version.cs]: Oh my bad, I misunderstood the diff, would you like me to add this back?

File: src/System.Runtime/tests/System/Version.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Runtime/tests/System/Version.cs]: > Would you like me to add them back?

Yes

File: src/System.Runtime/tests/System/Version.cs (hunk 2 above)
Comment (stephentoub) [inline src/System.Runtime/tests/System/Version.cs]: > would you like me to add this back?

Yes

File: src/System.Runtime/tests/System/Version.cs [hunk 4]
```diff
@@ -3,211 +3,263 @@
...
+        yield return new object[] { new Version(1, 2), new Version(1, 0, 1), 1 };
+        yield return new object[] { new Version(1, 2), new Version(1, 0, 0, 1), 1 };
+
+        yield return new object[] { new Version(1, 2, 3, 4), new Version(1, 2, 3, 4), 0 };
+        yield return new object[] { new Version(1, 2, 3, 4), new Version(1, 2, 3, 5), -1 };
+        yield return new object[] { new Version(1, 2, 3, 4), new Version(1, 2, 3, 3), 1 };
     }
 
-    private static void VerifyVersion(Version version, int major, int minor, int build, int revision)
+    [Theory, MemberData(nameof(CompareToTestData))]
```
Comment (bartonjs) [inline src/System.Runtime/tests/System/Version.cs]: Almost everywhere else you've used
```C#
[Theory]
[MemberData(nameof(SomeMethod))]
```

but here you used the one-line version.  Why is this one different?

File: src/System.Runtime/tests/System/Version.cs [hunk 5]
```diff
@@ -3,211 +3,263 @@
...
+    public static void TestCompareTo(Version version1, Version obj, int expected)
```
Comment (bartonjs) [inline src/System.Runtime/tests/System/Version.cs]: nit: `expected` => `expectedSign`.  Since that's all it's testing.

**Comment by bartonjs** [comment]
The new code looks good to me; but Stephen clearly had better eyes than me on lost tests for this one.

**Comment by hughbe** [comment]
Awesome, thanks for the comments folks: just pushed a PR feedback commit fixing all your comments

**Comment by stephentoub** [comment]
Thanks, @hughbe.
